lib/Controller/ChatController.php

```diff
@@ -434,11 +434,18 @@ public function receiveMessages(int $lookIntoFuture,
 
 		$i = 0;
 		$messages = $commentIdToIndex = $parentIds = [];
+		$historySince = $attendee->getHistorySince();
 		foreach ($comments as $comment) {
 			$id = (int) $comment->getId();
 			$message = $this->messageParser->createMessage($this->room, $this->participant, $comment, $this->l);
 			$this->messageParser->parseMessage($message);
 
+			if ($historySince) {
+				if ($message->getComment()->getCreationDateTime() < $historySince) {
```

Comment:
This makes an infinite request loop when scrolling too far up. In case the end is reached, we need to abort with the same return code as if there are no messages at all. See lines ~420+ with the `if (empty($comments)) {`

Reply:
Maybe instead of doing this here, you should add a array_filter method before that if statement.

Also we need to validate that `$lastKnownMessageId` is not older than the timestamp.

Reply:
I think that was solved in last commit.